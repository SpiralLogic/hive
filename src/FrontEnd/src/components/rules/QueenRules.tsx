import { Fragment, h } from 'preact';
import Hexagon from '../Hexagon';
import Hextille from '../Hextille';
import Row from '../Row';
import RuleCell from './RuleCell';

const QueenRules = () => (
  <>
    <Hextille class="rules">
      <Row>
        <RuleCell result="correct" />
        <RuleCell creature="ant" />
        <RuleCell creature="spider" />
        <Hexagon hidden={true} />
      </Row>
      <Row>
        <RuleCell result="correct" />
        <RuleCell
          zIndex={1}
          creature="queen"
          selected
          correctArrows={['topLeft', 'left', 'right', 'bottomRight']}
        />
        <RuleCell result="correct" />
        <RuleCell creature="spider" />
      </Row>
      <Row>
        <RuleCell creature="grasshopper" />
        <RuleCell result="correct" />
        <RuleCell creature="queen" />
        <RuleCell creature="beetle" />
      </Row>
      <Row>
        <Hexagon hidden={true} />
        <RuleCell creature="grasshopper" />
        <RuleCell creature="ant" />
        <RuleCell creature="beetle" />
      </Row>
    </Hextille>
    <caption>
      The queen can move one cell and must be palced by the 4th turn.<br/> No centre pieces can move until placed.
    </caption>
  </>
);

QueenRules.displayName = 'QueenRules';
export default QueenRules;
