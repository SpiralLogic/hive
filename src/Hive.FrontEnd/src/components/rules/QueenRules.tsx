import Hexagon from '../Hexagon';
import Hextille from '../Hextille';
import Row from '../Row';
import RuleCell from './RuleCell';
import { Rule } from './rule';

const QueenRule: Rule = {
  title: 'Queen',
  description: [
    'Can move one cell',
    'Must be in hive by 4th move',
    'No creatures in hive can move without queen',
  ],
  Rule: () => (
    <Hextille class="rules">
      <Row>
        <Hexagon hidden={true} />
        <RuleCell result="correct" />
        <RuleCell creature="ant" />
        <RuleCell creature="spider" />
      </Row>
      <Row>
        <RuleCell result="correct" />
        <RuleCell
          zIndex={1}
          creature="queen"
          selected
          correctArrows={['topLeft', 'left', 'right', 'bottomRight']}
        />
        <RuleCell result="correct" />
        <RuleCell creature="spider" />
      </Row>
      <Row>
        <Hexagon hidden={true} />
        <RuleCell creature="queen" />
        <RuleCell result="correct" />
        <RuleCell creature="beetle" />
      </Row>
      <Row>
        <RuleCell creature="grasshopper" />
        <RuleCell creature="ant" />
        <RuleCell creature="beetle" />
      </Row>
    </Hextille>
  ),
};

QueenRule.Rule.displayName = 'QueenRule';
export default QueenRule;
