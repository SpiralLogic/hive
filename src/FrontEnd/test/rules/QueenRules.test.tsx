import { h } from 'preact';
import { renderElement } from '../test-helpers';
import QueenRules from '../../src/components/rules/QueenRules';

describe('queen Rule snapshot tests', () => {
  it('snapshot', () => {
    const Rule = QueenRules.Rule;
    expect(renderElement(<Rule />)).toMatchSnapshot();
  });
});
