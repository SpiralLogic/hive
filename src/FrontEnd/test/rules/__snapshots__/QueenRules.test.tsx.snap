// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queen Rule snapshot tests snapshot 1`] = `
<body>
  <div>
    <div
      class="right"
      role="grid"
    >
      <div
        role="row"
      >
        <div
          role="none"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
        </div>
        <div
          class="correct"
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
            </svg>
            <span>
              ✓
            </span>
          </div>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile ant"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#ant"
              />
            </svg>
          </div>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile spider"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#spider"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        role="row"
      >
        <div
          class="correct"
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
            </svg>
            <span>
              ✓
            </span>
          </div>
        </div>
        <div
          role="cell"
          style="z-index: 4;"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile queen selected"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#queen"
              />
            </svg>
          </div>
          <svg
            class="arrow topLeft correct"
            style="--span: 1;"
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#arrow"
            />
          </svg>
          <svg
            class="arrow left correct"
            style="--span: 1;"
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#arrow"
            />
          </svg>
          <svg
            class="arrow right correct"
            style="--span: 1;"
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#arrow"
            />
          </svg>
          <svg
            class="arrow bottomRight correct"
            style="--span: 1;"
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#arrow"
            />
          </svg>
        </div>
        <div
          class="correct"
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
            </svg>
            <span>
              ✓
            </span>
          </div>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile spider"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#spider"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        role="row"
      >
        <div
          role="none"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile queen"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#queen"
              />
            </svg>
          </div>
        </div>
        <div
          class="correct"
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
            </svg>
            <span>
              ✓
            </span>
          </div>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile beetle"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#beetle"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        role="row"
      >
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile grasshopper"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#grasshopper"
              />
            </svg>
          </div>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile ant"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#ant"
              />
            </svg>
          </div>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile beetle"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#beetle"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
